git-hub

```diff
@@ -824,8 +829,37 @@ class CloneCmd (object):
         git_config('upstream', value=upstream)
         validate_url(remote_url, urltype)
         git('remote', 'add', '--', remote, remote_url)
-        infof('Fetching from {} ({})', remote, remote_url)
-        git_quiet(1, 'fetch', '--', remote)
+        # We also need to retry in here, although is less likely since we
+        # already spent some time doing the previous clone
+        cls.git_retry_if(args.triangular and forked,
+                'fetch', ['--', remote],
+                'Fetching from {} ({})'.format(remote, remote_url))
+
+    @classmethod
+    def git_retry_if(cls, condition, cmd, args, progress_msg):
+        # If we are not retrying, just do it once
+        if not condition:
+            infof(progress_msg)
+            git_quiet(1, cmd, *args)
+            return
+        # ~5m total wait time with "human" exponential backoff
+        wait_times = [1, 2, 5, 10, 15, 30, 60, 90, 90]
+        retries = 0
+        while retries < len(wait_times):
+            try:
+                infof(progress_msg)
+                git_quiet(1, cmd, *args)
+                break
+            except GitError as e:
+                t = wait_times[retries]
+                warnf("Couldn't {}, maybe GitHub is still performing "
+                        "the fork. Retrying in {}s ({})", cmd, t, e)
+                time.sleep(t)
+                retries += 1
+        else:
+            die("Couldn't {} after waiting {}m in {} retries. Maybe it's "
+                "time to contact GitHub's support: https://github.com/contact",
+                    cmd, sum(wait_times)/60.0, retries)
```

Comment:
Mmm, good point. OK, I'll use `//`, I don't think this is important enough to implement a full human-readable time format with minutes and seconds.